packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -13,21 +13,25 @@ typedef constructor<T> = T Function(
 /// The [SerializableEntity] is the base class for all serializable objects in
 /// Serverpod, except primitives.
 abstract mixin class SerializableEntity {
-  /// Returns a serialized JSON structure of the model, ready to be sent
-  /// through the API. This does not include fields that are marked as
-  /// database only.
-  dynamic toJson();
-
   /// Returns a serialized JSON structure of the model which also includes
   /// fields used by the database.
-  dynamic allToJson() => toJson();
+  dynamic toJson();
 
   @override
   String toString() {
     return SerializationManager.encode(this);
   }
 }
 
+/// The [ConstraintSerialization] defines a toJson method which makes it
+/// possible to limit what fields are serialized
+abstract interface class ConstraintSerialization {
+  /// Returns a serialized JSON structure of the model, ready to be sent
+  /// through the API. This does not include fields that are marked as
+  /// database only.
```

Comment:
Suggestion: this description does not need to talk about what fields are included or not. Just mention what the serialization is used for.